src/hooks/screen/useChatScreenHook.ts

```diff
@@ -124,17 +127,62 @@ function useChatScreenHook(type: 'SIGNED' | 'ANONYMOUS'): UseChatScreenHook {
       }
 
       let response;
+
+      const allAttachmentPromises: Promise<void>[] = attachments.map(async (item: any) => {
+        if (item.type === 'image') {
+          const uploadedImageUrl = await ImageUtils.uploadImage(item.asset_url);
+          return {
+            ...item,
+            asset_url: uploadedImageUrl.data.url,
+            thumb_url: uploadedImageUrl.data.url
+          };
+        }
+        if (item.type === 'video') {
+          const uploadedImageUrl = await ImageUtils.uploadImage(item.asset_url);
+          const uploadedUrl = await ImageUtils.uploadFile(
+            item.video_path,
+            item.video_name,
+            item.video_type
+          );
+          return {
+            ...item,
+            asset_url: uploadedImageUrl.data.url,
+            thumb_url: uploadedImageUrl.data.url,
+            video_path: uploadedUrl.data.url
+          };
+        }
+        if (item.type === 'file') {
+          const uploadedUrl = await ImageUtils.uploadFile(
+            item.file_path,
+            item.file_name,
+            item.file_type
+          );
+          return {
+            ...item,
+            asset_url: uploadedUrl.data.url,
+            thumb_url: uploadedUrl.data.url,
+            file_path: uploadedUrl.data.url
+          };
+        }
+
+        return item;
+      });
+
+      const newAttachments = await Promise.all(allAttachmentPromises);
+
       if (type === ANONYMOUS) {
         response = await AnonymousMessageRepo.sendAnonymousMessage(
           selectedChannel?.id,
           message,
-          replyData?.id
+          replyData?.id,
+          newAttachments
         );
       } else {
         response = await SignedMessageRepo.sendSignedMessage(
           selectedChannel?.id,
           message,
           channelType,
+          newAttachments,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `sendAnonymousMessage` and `sendSignedMessage` functions now include an `attachments` parameter. This is a necessary change to send attachments along with the chat message.
```diff
+          response = await AnonymousMessageRepo.sendAnonymousMessage(
+            selectedChannel?.id,
+            message,
+           replyData?.id,
+           newAttachments
+          );
+        } else {
+          response = await SignedMessageRepo.sendSignedMessage(
+            selectedChannel?.id,
+            message,
+            channelType,
+           newAttachments,
+           replyData?.id
+         );
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->